lib/fs.js

```diff
@@ -2008,31 +2008,55 @@ function ReadStream(path, options) {
   this.bytesRead = 0;
   this.closed = false;
 
-  if (this.start !== undefined) {
-    if (typeof this.start !== 'number' || Number.isNaN(this.start)) {
-      throw new ERR_INVALID_ARG_TYPE('start', 'number', this.start);
-    }
-    if (this.end === undefined) {
-      this.end = Infinity;
-    } else if (typeof this.end !== 'number' || Number.isNaN(this.end)) {
-      throw new ERR_INVALID_ARG_TYPE('end', 'number', this.end);
-    }
+  // Case 0: If either start or end is undefined, set them to defaults
+  // (0, Infinity) respectively.
+  this.start = this.start === undefined ? 0 : this.start;
```

Comment:
I’m mostly sure this is the cause of the test failure – `start: 0` and `start: undefined` aren’t equivalent, because the former means that the stream will use reads with an `offset` parameter (which doesn’t work on all types of files). If `start: undefined` was passed, this value should be `undefined`, too.

Reply:
Alright, let me just remove the condition then.